tests/integration_tests_rust.rs

```diff
@@ -1969,21 +1968,41 @@ async fn lsps2_client_trusts_lsp() {
 	println!("Paying JIT invoice!");
 	let payment_id = payer_node.bolt11_payment().send(&res, None).unwrap();
 	println!("Payment ID: {:?}", payment_id);
-	expect_channel_pending_event!(service_node, client_node.node_id());
+	let funding_txo = expect_channel_pending_event!(service_node, client_node.node_id());
 	expect_channel_ready_event!(service_node, client_node.node_id());
 	expect_channel_pending_event!(client_node, service_node.node_id());
 	expect_channel_ready_event!(client_node, service_node.node_id());
+
+	// Check the fundign transaction hasn't broadcasted yet and nodes aren't seeing it.
 	println!("Try to find funding tx... It won't be found yet, as the client has not claimed it.");
-	let mut funding_tx_found = false;
-	for _ in 0..50 {
-		std::thread::sleep(std::time::Duration::from_millis(100));
-		let current_mempool = bitcoind.client.get_raw_mempool().unwrap();
-		if current_mempool.0.len() > initial_mempool_size {
-			funding_tx_found = true;
-			break;
-		}
+	for _ in 0..30 {
+		tokio::time::sleep(std::time::Duration::from_millis(100)).await;
```

Comment:
In the happy flow, we are always waiting 3s I think. Is a loop necessary then? Maybe just delay 3s and check.

Reply:
Done.